crates/runner-nordic/src/tasks/radio.rs

```diff
@@ -0,0 +1,68 @@
+// Copyright 2023 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use rubble::bytes::ByteWriter;
+use rubble::link::{NextUpdate, RadioCmd};
+use rubble::time::{Duration, Timer};
+use wasefire_board_api::radio::Api;
+use wasefire_board_api::{Error, Support};
+use wasefire_logger as logger;
+
+use crate::with_state;
+
+pub enum Impl {}
+
+impl Api for Impl {
+    fn enable() -> Result<(), Error> {
+        with_state(|state| {
+            let scanner_cmd =
+                state.ble_scanner.configure(state.ble_timer.now(), Duration::millis(500));
+            state.ble_radio.configure_receiver(scanner_cmd.radio);
+            state.ble_timer.configure_interrupt(scanner_cmd.next_update);
+            Ok(())
+        })
+    }
+
+    fn disable() -> Result<(), Error> {
+        with_state(|state| {
+            state.ble_timer.configure_interrupt(NextUpdate::Disable);
+            state.ble_radio.configure_receiver(RadioCmd::Off);
+            Ok(())
+        })
+    }
+
+    fn read(output: &mut [u8]) -> Result<usize, Error> {
+        with_state(|state| {
+            let packet = state.ble_packet_queue.pop_front().ok_or(Error::World)?;
+            let len = packet.len();
```

Comment:
I guess this is where `len()` is used and it seems it's only for debugging. I'll have to think a bit about what's the best option. For now, let's keep it like that.